src/corelibs/U2View/src/ov_sequence/sequence_info/SequenceInfo.cpp

```diff
@@ -533,6 +552,22 @@ bool SequenceInfo::eventFilter(QObject* object, QEvent* event) {
     return false;
 }
 
+void SequenceInfo::statisticLabelLinkActivated(const QString& link) {
+    if (link == CAPTION_SEQ_MELTING_TEMPERATURE) {
+        auto dialog = new TempCalcDialog(annotatedDnaView->getActiveSequenceWidget(), temperatureCalculator->getSettings());
```

Comment:
I use this way because Qt developers notes, that this solution is better. This is a part of [exec()](https://doc-snapshots.qt.io/qt5-5.15/qdialog.html#exec) documentation:

![image](https://user-images.githubusercontent.com/26257527/214796296-770cd42c-445d-4612-83a0-54058cb85bde.png)

And, by the way, we already faced bugs, which appeared because of this problem (e.g. see [UGENE-7549](https://ugene.dev/tracker/browse/UGENE-7549)). I think, the argument that I need to rewrite it to `exec()` because we have it everywhere isn't that good, it just means that we have been using the incorrect solution for the whole time.

Reply:
You can create another bug that will focus on the problem like
1) Rewrite 100+ places of using ->exec()
2) Rewrite 100+ places of using QMessageBox

This is not related to Primer3 and requires additional discussion.

About fix for UGENE-7549: that was a bug in Qt 5.12.x but not in UGENE.
Now with Qt 5.15  that code was already migrated to the standard way:  using ->exec

I suggest removing this question from the Primer3 PR discussion: I want to proceed with approval and this unrelated issue will block it.

Reply:
What if I say "No"? Because nothing convincing I read from above, just common phrases, which means in generally, "I'm a teamleader, so you should do what I said"